File: gym/static/js/employee.js
$(()=>{
    $("#select").click(
        ()=>{
            $("#table").empty();
           $.post(
               '/employee_select',
               {
                   eid:$("#eid").val()
               },
               (json)=>{
                   th="<tr>\n" +
                       "<th>Employee ID</th>\n" +
                       "<th>UserName</th>\n" +
                       "<th>Email</th>\n" +
                       "<th>Phone</th>\n" +
                       "<th>Salary</th>\n" +
                       "<th>Is_internship</th>\n"+
                       "</tr>";
                   $("#table").append(th);
                   json.forEach(
                       (row)=>{
                            $("#table").append(
                                "<tr><td>"+row.eid+"</td><td>"+
                                row.username+"</td><td>"+row.email+"</td><td>"+
                                row.phone+"</td><td>"+row.salary+"</td><td>"+
                                row.is_internship+"</td></tr>"
                            )
                       }
                   );
               }
           )
        }
    );

$("#delete").click(
        ()=>{
            if($("#eid").val()==""){
                alert("please input eid to delete")
            }
            else {
                $.getJSON(
                '/employee_delete',
                {
                    eid:$("#eid").val()
                },
                (json)=>{
                    if(json.status==1)
                        alert("delete success!")
                    else if(json.status==0)
                        alert("delete fail ! please try again")
                    else
                        alert("unknown error, please try again !")

                }

            )
            }
        }
    );

$("#update").click(
        ()=>{
            if($("#eid").val()==""&& $("#username").val()==""&& $("#email").val()==""&& $("#phone").val()=="" && $("#salary").val()=="" && $("#is_internship").val()=="" && $("#password").val()==""){
                 alert("please input content!")
            }
            $.getJSON(
                '/employee_update',
                {
                    eid:$("#eid").val(),
                    username:$("#username").val(),
                    email:$("#email").val(),
                    phone:$("#phone").val(),
                    salary:$("#salary").val(),
                    is_internship:$("#is_internship").val(),
                    password:$("#password").val()
                },
                (json)=>{
                    if(json.status==1)
                        alert("update success!")
                    if(json.status==0)
                        alert("update fail please use correct employee ID")
                }
            )
        }
    );


})

File: gym/views/employee.py
from gym import app
from flask import render_template, request, jsonify
from gym.daos.dao_employee import DaoEmployee


@app.route('/employee_select', methods=['POST'])
def employee_select():
    eid = request.form.get('eid')
    dao_employee = DaoEmployee()
    if eid is None or eid == '':
        all_rows = dao_employee.find_all()
    else:
        all_rows = dao_employee.find_employee(int(eid))
    return jsonify(all_rows)


@app.route('/employee_delete', methods=['GET'])
def employee_delete():
    eid=int(request.args.get('eid'))
    dao_employee = DaoEmployee()
    all_rows=dao_employee.find_all()
    status=0
    for row in all_rows:
        if eid==int(row['eid']):
            dao_employee.delete(eid)
            status=1
            break
        else:
            continue
    else:
        status=0

    return jsonify(status=status)


@app.route('/employee_update', methods=['GET'])
def employee_update():
    eid=int(request.args.get('eid'))
    username=request.args.get('username')
    email=request.args.get('email')
    phone=request.args.get('phone')
    salary=int(request.args.get('salary'))
    is_internship=int(request.args.get('is_internship'))
    password=request.args.get('password')
    dao_employee = DaoEmployee()
    all_rows = dao_employee.find_all()
    for row in all_rows:
        if eid==int(row['eid']):
            dao_employee.update(email,phone,salary,password,is_internship,username,eid)
            status=1
            break
        else:
            continue
    else:
        status=0

    return jsonify(status=status)


